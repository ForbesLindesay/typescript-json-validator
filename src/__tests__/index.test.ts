import run from '../';

// let validate: any;

test('run', () => {
  run(['src/Example.ts', 'ExampleType']);
  // validate = require('../Example.validator').default;
});

// test('valid', () => {
//   expect(
//     validate({
//       value: 'Hello World',
//     }),
//   ).toMatchInlineSnapshot(`
// Object {
//   "answer": 42,
//   "value": "Hello World",
// }
// `);
//   expect(
//     validate({
//       value: 'Hello World',
//       email: 'forbes@lindesay.co.uk',
//     }),
//   ).toMatchInlineSnapshot(`
// Object {
//   "answer": 42,
//   "email": "forbes@lindesay.co.uk",
//   "value": "Hello World",
// }
// `);
// });

// test('invalid', () => {
//   expect(() => validate({})).toThrowErrorMatchingInlineSnapshot(`
// "ExampleType should have required property 'value'
// ExampleType value:

// { answer: 42 }"
// `);
//   expect(() =>
//     validate({
//       value: 'Hello World',
//       email: 'forbeslindesay.co.uk',
//     }),
//   ).toThrowErrorMatchingInlineSnapshot(`
// "ExampleType.email should match format \\"email\\"
// ExampleType value:

// { value: 'Hello World',
//   email: 'forbeslindesay.co.uk',
//   answer: 42 }"
// `);
// });
