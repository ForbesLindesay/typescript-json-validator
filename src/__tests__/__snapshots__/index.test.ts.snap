// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic run tests collection invalid 1`] = `"Invalid ExampleType: ExampleType must have required property 'value'"`;

exports[`Basic run tests collection invalid 2`] = `"Invalid ExampleType: ExampleType must have required property 'value', ExampleType/email must match format \\"email\\""`;

exports[`Basic run tests collection valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Basic run tests collection valid 2`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Basic run tests collection with separate schema file invalid 1`] = `"Invalid ExampleType: ExampleType must have required property 'value'"`;

exports[`Basic run tests collection with separate schema file invalid 2`] = `"Invalid ExampleType: ExampleType must have required property 'value', ExampleType/email must match format \\"email\\""`;

exports[`Basic run tests collection with separate schema file valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Basic run tests collection with separate schema file valid 2`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Basic run tests invalid 1`] = `
"ExampleType must have required property 'value'

{ answer: 42 }"
`;

exports[`Basic run tests invalid 2`] = `
"ExampleType must have required property 'value', ExampleType/email must match format \\"email\\"

{ email: 'forbeslindesay.co.uk', answer: 42 }"
`;

exports[`Basic run tests valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Basic run tests valid 2`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Basic run tests with cleaner invalid 1`] = `
"ExampleType must have required property 'value'

{ answer: 42 }"
`;

exports[`Basic run tests with cleaner invalid 2`] = `
"ExampleType must have required property 'value', ExampleType/email must match format \\"email\\"

{ email: 'forbeslindesay.co.uk', answer: 42 }"
`;

exports[`Basic run tests with cleaner valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Basic run tests with cleaner valid 2`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Basic run tests with separate schema file invalid 1`] = `
"ExampleType must have required property 'value'

{ answer: 42 }"
`;

exports[`Basic run tests with separate schema file invalid 2`] = `
"ExampleType must have required property 'value', ExampleType/email must match format \\"email\\"

{ email: 'forbeslindesay.co.uk', answer: 42 }"
`;

exports[`Basic run tests with separate schema file valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Basic run tests with separate schema file valid 2`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Permissive run tests collection Clean Extra 1`] = `
Object {
  "answer": 1,
  "value": "ddd",
}
`;

exports[`Permissive run tests collection Clean Extra 2`] = `"Invalid ExampleType: ExampleType must NOT have additional properties"`;

exports[`Permissive run tests collection invalid 1`] = `"Invalid ExampleType: ExampleType must have required property 'value'"`;

exports[`Permissive run tests collection invalid 2`] = `"Invalid ExampleType: ExampleType must have required property 'value'"`;

exports[`Permissive run tests collection invalid 3`] = `"Invalid ExampleType: ExampleType must have required property 'value', ExampleType/email must match format \\"email\\""`;

exports[`Permissive run tests collection invalid 4`] = `"Invalid ExampleType: ExampleType must have required property 'value', ExampleType/email must match format \\"email\\""`;

exports[`Permissive run tests collection valid 1`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Permissive run tests collection valid 2`] = `
Object {
  "answer": 42,
  "value": "Hello World",
}
`;

exports[`Permissive run tests collection valid 3`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;

exports[`Permissive run tests collection valid 4`] = `
Object {
  "answer": 42,
  "email": "forbes@lindesay.co.uk",
  "value": "Hello World",
}
`;
